Hw4/Hw4/Routers.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Murat Khamatyanov. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Hw4;
+
+/// <summary>
+/// class for maximum spanning tree function.
+/// </summary>
+public class Routers
+{
+    /// <summary>
+    /// maximum spanning tree function.
+    /// </summary>
+    /// <param name="pathInput"> path to input file. </param>
+    /// <param name="pathOutput"> path to output file. </param>
+    /// <exception cref="ArgumentNullException"> if null input data. </exception>
+    /// <exception cref="ArgumentException"> if input graph is disconnected. </exception>
+    public static void Optimum(string pathInput, string pathOutput)
+    {
+        var graph = new Dictionary<int, Dictionary<int, int>>();
+        var sr = new StreamReader(pathInput);
```

Comment:
```suggestion
        using var sr = new StreamReader(pathInput);
```
Чтобы оно само закрылось при возникновении исключения